File: Sort/insertion sort/insertion.py
import random

def insertion_sort(arr):
    for i in range(len(arr)):
        for j in range(i):
            if (arr[j] > arr[i]):
                tmp = arr[i]
                for k in range(i, j, -1):
                    arr[k] = arr[k - 1]
                arr[j] = tmp;
                break;
    return arr

def pr_arr(arr, string):
    print("\n\t\t%s\n\t\t" % string, end="")
    for i in range(len(arr)):
        print("%3d" % (arr[i]), end='');
        if (i != 0 and i % 10 == 0):
            print("\n\t\t", end="");
    print("")

if __name__ == "__main__":
    arr = [random.randint(0, 9) for i in range(150)]
    pr_arr(arr, "Initialized");
    pr_arr(insertion_sort(arr), "After insertion_sort")

File: Sort/insertion sort/insertion.c
#include <stdio.h>
#include <time.h>
#include <stdlib.h>
#define SIZE 150

void insertion_sort(int *arr, int size)
{
		int tmp, j;
		if (size > 1)
		{
				for (int i = 1; i < size; i++)
				{
						j = i - 1;
						while (arr[j] > arr[j + 1] && j >= 0)
						{
								tmp = arr[j];
								arr[j] = arr[j + 1];
								arr[j + 1] = tmp;
								--j;
						}
				}
		}
}

void pr_arr(int *arr, const char *str, int size)
{
		printf("\n\t\tsize: %d, %s\n\t\t", size, str);
		for (int i = 0; i < size; i++)
		{
				printf("%3d ", arr[i]);
				if (i != 0 && i % 10 == 0)
						printf("\n\t\t");
		}
		printf("\n");
}

int main(void)
{
		srand(time(NULL));
		int arr[SIZE];
		for (int i = 0; i < SIZE; i++)
				arr[i] = rand() % 10;
		pr_arr(arr, "Iinitialized", SIZE);
		insertion_sort(arr, SIZE);
		pr_arr(arr, "After Insertion sort", SIZE);
		return(0);
}
